--- a/node_modules/yaml-language-server/out/server/src/languageservice/yamlLanguageService.d.ts	2024-05-02 15:33:31
+++ b/node_modules/yaml-language-server/out/server/src/languageservice/yamlLanguageService.d.ts	2024-05-02 15:34:03
@@ -25,6 +25,7 @@
 export interface LanguageSettings {
     validate?: boolean;
     hover?: boolean;
+    hoverSettings?: {showSource?: boolean; showTitle?: boolean;};
     completion?: boolean;
     format?: boolean;
     isKubernetes?: boolean;
